pkg/api/api.go

```diff
@@ -51,14 +52,18 @@ func New(serverIDFunc func() string) *API {
 // ServerID returns the Server ID to be used in responses. A custom server ID
 // generation function can be passed to New().
 func (a API) ServerID() string {
+	serverID := "<unknown>"
 	if a.serverIDFunc != nil {
-		return a.serverIDFunc()
+		serverID = a.serverIDFunc()
+	} else {
+		if hn, err := os.Hostname(); err == nil {
+			serverID = hn
+		}
 	}
-	hn, err := os.Hostname()
-	if err != nil {
-		return "<unknown>"
+	if err := limits.Validator.ValidateServerID(serverID); err != nil {
+		panic(err)
```

Comment:
definitely not a fan of panicking in library code - how complex is it to change `ServerID`'s signature instead?

Reply:
I agree, but it's not clear, where ServerID() can be called. It looks like "general getter". Imho it would be better to calculate the serverId explicitly in some initialisation code, validate it and then just return to every call. In order to do it immutable, we can warp in in lambda. What do you think?